Error when installing on Windows.

I get this error when i try to install the module via pip. 

```
>pip install numba-kdtree
Collecting numba-kdtree
  Using cached numba-kdtree-0.1.5.tar.gz (23 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numba>=0.52 in c:\users\user01\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from numba-kdtree) (0.56.4)
Requirement already satisfied: llvmlite<0.40,>=0.39.0dev0 in c:\users\user01\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from numba>=0.52->numba-kdtree) (0.39.1)
Requirement already satisfied: numpy<1.24,>=1.18 in c:\users\user01\appdata\local\packages\pythonsoftwarefoundation.python.3.10_qbz5n2kfra8p0\localcache\local-packages\python310\site-packages (from numba>=0.52->numba-kdtree) (1.23.5)
Requirement already satisfied: setuptools in c:\program files\windowsapps\pythonsoftwarefoundation.python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\lib\site-packages (from numba>=0.52->numba-kdtree) (65.5.0)
Building wheels for collected packages: numba-kdtree
  Building wheel for numba-kdtree (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for numba-kdtree (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [96 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\numba_kdtree
      copying numba_kdtree\ckdtree_ctypes.py -> build\lib.win-amd64-cpython-310\numba_kdtree
      copying numba_kdtree\kd_tree.py -> build\lib.win-amd64-cpython-310\numba_kdtree
      copying numba_kdtree\_version.py -> build\lib.win-amd64-cpython-310\numba_kdtree
      copying numba_kdtree\__init__.py -> build\lib.win-amd64-cpython-310\numba_kdtree
      running build_ext
      building 'numba_kdtree._ckdtree' extension
      creating build\temp.win-amd64-cpython-310
      creating build\temp.win-amd64-cpython-310\Release
      creating build\temp.win-amd64-cpython-310\Release\src
      creating build\temp.win-amd64-cpython-310\Release\src\ckdtree
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\include" 
-Isrc\ckdtree "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\include" "-IC:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\Include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files 
(x86)\Windows Kits\10\\include\10.0.19041.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tpsrc\ckdtree\build.cpp /Fobuild\temp.win-amd64-cpython-310\Release\src\ckdtree\build.obj /O2 /DKDTREE_COMPILING=1
      build.cpp
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(39): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(48): note: see reference to class template instantiation 'ckdtreenode<T>' being compiled
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(41): error C3646: 'split_dim': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(41): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(42): error C3646: 'children': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(42): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(44): error C3646: 'start_idx': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(44): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(45): error C3646: 'end_idx': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(45): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(46): error C3646: '_less': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(46): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(47): error C3646: '_greater': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(47): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(61): error C3646: 'n': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(68): note: see reference to class template instantiation 'ckdtree<T>' being compiled
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(61): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(62): error C3646: 'm': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(62): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(63): error C3646: 'leafsize': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(63): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(66): error C2143: syntax error: missing ';' before '*'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(66): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(66): error C2238: unexpected token(s) preceding ';'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(67): error C3646: 'size': unknown override specifier
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree_decl.h(67): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(39): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(42): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(51): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(55): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): error C2146: syntax error: missing ';' before identifier 'ckdtree_size_float'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(60): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(60): error C2086: 'int ssize_t': redefinition
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): note: see declaration of 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(60): error C2146: syntax error: missing ';' before identifier 'ckdtree_size_double'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(60): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(63): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(67): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(70): error C2923: 'std::vector': 'ssize_t' is not a valid template type argument for parameter '_Ty'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): note: see declaration of 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(70): error C2976: 'std::vector': too few template arguments
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include\vector(480): note: see declaration of 'std::vector'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(70): warning C4099: 'std::vector': type name first seen using 'class' now seen using 'struct'
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include\vector(480): note: see declaration of 'std::vector'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(70): error C2990: 'std::vector': non-class template has already been declared as a class template
      C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include\vector(480): note: see declaration of 'std::vector'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(73): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(77): error C2061: syntax error: identifier 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(80): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(80): error C2086: 'int ssize_t': redefinition
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(58): note: see declaration of 'ssize_t'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(80): error C2146: syntax error: missing ';' before identifier 'radius_result_set_get_size'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(80): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(82): error C2061: syntax error: identifier 'ssize_t'
      src\ckdtree\build.cpp(21): error C2447: '{': missing function header (old-style formal list?)
      src\ckdtree\build.cpp(195): error C2061: syntax error: identifier 'ssize_t'
      src\ckdtree\build.cpp(201): error C2065: 'build': undeclared identifier
      src\ckdtree\build.cpp(201): error C2062: type 'float' unexpected
      src\ckdtree\build.cpp(202): error C2039: 'size': is not a member of 'ckdtree<float>'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(35): note: see declaration of 'ckdtree<float>'
      src\ckdtree\build.cpp(206): error C2061: syntax error: identifier 'ssize_t'
      src\ckdtree\build.cpp(212): error C2065: 'build': undeclared identifier
      src\ckdtree\build.cpp(212): error C2062: type 'double' unexpected
      src\ckdtree\build.cpp(213): error C2039: 'size': is not a member of 'ckdtree<double>'
      C:\Users\user01\AppData\Local\Temp\pip-install-w3klou04\numba-kdtree_7207746e8de74428a1142773e761e6c6\src\ckdtree\ckdtree.h(36): note: see declaration of 'ckdtree<double>'
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for numba-kdtree
Failed to build numba-kdtree
ERROR: Could not build wheels for numba-kdtree, which is required to install pyproject.toml-based projects
```


And this one when trying the "from source" Installation as administrator:

```
> python setup.py install
running install
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\lib\site-packages\setuptools\command\easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
error: can't create or remove files in install directory

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 13] Permission denied: 'C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\\Lib\\site-packages\\test-easy-install-10332.write-test'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\Lib\site-packages\

Perhaps your account does not have write access to this directory?  If the
installation directory is a system-owned directory, you may need to sign in
as the administrator or "root" account.  If you do not have administrative
access to this machine, you may wish to choose a different installation
directory, preferably one that is listed in your PYTHONPATH environment
variable.

For information on other options, you may wish to consult the
documentation at:

  https://setuptools.pypa.io/en/latest/deprecated/easy_install.html

Please make the appropriate changes for your system and try again.
```

The Installation on Linux works. But using it on Windows would be my prefered choice. Any help would be appreciated.

--- a/src/ckdtree/ckdtree_decl.h
+++ b/src/ckdtree/ckdtree_decl.h
@@ -31,6 +31,12 @@
     #endif
 #endif
 
+// define the ssize_t type for windows systems
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 
 #define ckdtree_intp_t ssize_t
 

